courseraresearchexports/exports/utils.py

```diff
@@ -116,88 +94,3 @@ def lookup_partner_short_name_by_id(partner_id):
     response = requests.get(requests.compat.urljoin(PARTNER_API, partner_id))
```

Comment:
Added it in.  I had remove usages of courseSlug/partnerShortName to id conversions, but they could still be useful for anyone extending the library.  

I don't think there's pythonic way to abstract this pattern. I don't think you usually create multiline anonymous functions, so it'd be naming each function and passing it to an error handling decorator.